src/main/groovy/com/palantir/gradle/shadowjar/ComposableManifestAppenderTransformer.groovy

```diff
@@ -0,0 +1,89 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Change: Different package, added imports
+package com.palantir.gradle.shadowjar
+
+import com.github.jengelman.gradle.plugins.shadow.transformers.Transformer
+import com.github.jengelman.gradle.plugins.shadow.transformers.TransformerContext
+import shadow.org.apache.tools.zip.ZipOutputStream
+import shadow.org.apache.tools.zip.ZipEntry
+import shadow.org.codehaus.plexus.util.IOUtil
+import org.gradle.api.file.FileTreeElement
+
+import static java.nio.charset.StandardCharsets.*
+import static java.util.jar.JarFile.*
+
+// Originally taken from https://github.com/johnrengelman/shadow/blob/6.1.0/src/main/groovy/com/github/jengelman/
+// gradle/plugins/shadow/transformers/ManifestAppenderTransformer.groovy
```

Comment:
surely we should just be PRing this upstream to mr johnrengelman?

Reply:
I'm happy to do this if you think that makes sense, though was a bit concerned as to timescales since the last thing was merged in early October!

Reply:
I'll PR this upstream, but also think we should get this through here as we need it for atlasdb-proxy's migrations and I don't know how actively the shadow plugin is maintained (as mentioned, the last time anything was merged was in early October, and there is a 26 day old PR that has seen no response)